src/subl.ts

```diff
@@ -1,10 +1,83 @@
+const generateProjects: Fig.Generator = {
+  template: "filepaths",
+  filterTemplateSuggestions: (suggestions) => {
+    const out: Fig.Suggestion[] = [];
+    for (const suggestion of suggestions) {
+      if (suggestion.type === "folder") {
+        out.push(suggestion);
+        continue;
+      }
+      if (/^\.sublime-(project|workspace)$/.test(suggestion.name)) {
+        suggestion.priority = 75;
+        out.push(suggestion);
+      }
+    }
+    return out;
+  },
```

Comment:
I would prefer a cleaner solution, something like:
```suggestion
  filterTemplateSuggestions: (suggestions) => suggestions.map(suggestion => {
    if (suggestion.type === "folder") {
      return suggestion;
    }

    return {
      ...suggestion,
      priority: /^\.sublime-(project|workspace)$/.test(suggestion.name)
        ? 75
        : suggestion.priority
    }
  })
```

Reply:
I think the actual problem here was my lack of comments explaining the intention :)
This loop also removes the files that _don't_ match the regex, so it's a combined filter/map. The equivalent would be this, which is slower. Debatably clearer, but comments can fix that in the imperative version.

```typescript
const generateProjects: Fig.Generator = {
  template: "filepaths",
  filterTemplateSuggestions: (suggestions) =>
    suggestions
      .filter((suggestion) => {
        if (suggestion.type === "folder") {
          return true;
        }
        return /^\.sublime-(project|workspace)$/.test(suggestion.name);
      })
      .map((suggestion) => {
        if (suggestion.type === "folder") {
          return suggestion;
        }
        return { ...suggestion, priority: 75 };
      }),
};
```

Lmk if you'd rather go ahead with this version!

Reply:
That would be awesome if you could add some comments!